### PR TITLE
Updated SGF loading

### LINKED ISSUE #51: Refactor SGF conversions?
@ttrine and I never fully agreed on whether the `GameState` to/from SGF functionality would be part of the `GameState` class itself or reside in `game_converter`. Maybe the most reasonable thing to do would be to refactor the conversions entirely (i.e. some `state_to_sgf` and `sgf_to_state` in a separate file).

### LINKED ISSUE #53: consider switching to jtauber/sgf for sgf library
Just creating an issue here for discussion (close if it's a non-starter).

The code base currently has an old SGF parsing library that's LGPL and from 2000.

If you are at all interested in a pip installable, MIT-licensed SGF parsing library that will be actively maintained by someone already contributing to this repo, I'd be delight to make sure https://github.com/jtauber/sgf is up to the task.

### PR DESCRIPTION
**Pull request by wrongu**
Removing `sgflib` and replacing it with [jtauber/sgf](/jtauber/sgf). SGF handling moved to a separate file `AlphaGo/util.py`

This closes #53, and fixes #51 . Still needs more test cases as per #59 

@ttrine there are conflicts with the `training` branch; the merge needs to be done carefully, which is why I'm making a PR and not just pushing to `develop`.

### CODE REVIEW

File: AlphaGo/util.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	"""Iterates over (GameState, move, player) tuples in the first game of the given SGF file.
+
+	Ignores variations - only the main line is returned.
+
+	Note that the final tuple returned is the penultimate state,
+	but because 'gs' is modified in-place the state is at the final
+	position after iteration completes. See sgf_to_gamestate
+	"""
+	collection = sgf.parse(sgf_string)
+	game = next(iter(collection))
```
Comment (jtauber) [inline AlphaGo/util.py]: just out of interest, why do you do this rather than just `game = collection[0]` ?

File: AlphaGo/util.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	node_iterator = iter(game)
+	root = next(node_iterator)
+	gs = _sgf_init_gamestate(root)
+	for node in node_iterator:
```
Comment (jtauber) [inline AlphaGo/util.py]: I'm now wondering if my library should support grabbing the root and the rest more easily so these four lines could be replaced by something like:

```
gs = _sgf_init_gamestate(game.root)
for node in game.rest:
```

File: AlphaGo/util.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import sgf
+import string
+from AlphaGo import go
+
+def _parse_sgf_move(node_value):
```
Comment (jtauber) [inline AlphaGo/util.py]: I wonder if my library should just provide something like this

File: AlphaGo/util.py (hunk 1 above)
Comment (wrongu) [inline AlphaGo/util.py]: because I didn't remember I could do that...

Maybe it should also support `__len__` if it supports `__getitem__`

**Pull request merged**